### PR TITLE
Add the script for cmd line Tab completion

### PR DESCRIPTION
**Pull request by RNAer**
`source autocomplete.sh` will enable that.

### CODE REVIEW

**Comment by RNAer** [comment]
This is ready for review. I changed the `click` framework so we can have both 2-level and 3-level nested subcommands freely. Was too lazy to separate it out as another PR...

File: autocomplete.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+_micronota_completion() {
+    COMPREPLY=( $( env COMP_WORDS="${COMP_WORDS[*]}" \
+                   COMP_CWORD=$COMP_CWORD \
+                   _MICRONOTA_COMPLETE=complete $1 ) )
+    return 0
+}
+
+complete -F _micronota_completion -o default micronota;
```
Comment (mortonjt) [inline autocomplete.sh]: Should this be automatically sourced in `setup.py`?

**Comment by mortonjt** [comment]
:+1:

**Comment by mortonjt** [comment]
I'll open up an issue about the autocomplete.sh

**Pull request merged**